plugins/inputs/elasticsearch/elasticsearch.go

```diff
@@ -460,6 +483,87 @@ func (e *Elasticsearch) gatherClusterStats(url string, acc telegraf.Accumulator)
 	return nil
 }
 
+func (e *Elasticsearch) gatherIndicesStats(url string, acc telegraf.Accumulator) error {
+	indicesStats := &struct {
+		Shards  map[string]interface{} `json:"_shards"`
+		All     map[string]interface{} `json:"_all"`
+		Indices map[string]indexStat   `json:"indices"`
+	}{}
+
+	if err := e.gatherJSONData(url, indicesStats); err != nil {
+		return err
+	}
+	now := time.Now()
+
+	// Total Shards Stats
+	shardsStats := map[string]interface{}{}
+	for k, v := range indicesStats.Shards {
+		shardsStats[k] = v
+	}
+	acc.AddFields("elasticsearch_indicestats_shards_total", shardsStats, map[string]string{"name": ""}, now)
+
+	// All Stats
+	for m, s := range indicesStats.All {
+		// parse Json, ignoring strings and bools
+		jsonParser := jsonparser.JSONFlattener{}
+		err := jsonParser.FullFlattenJSON("_", s, true, true)
+		if err != nil {
+			return err
+		}
+		acc.AddFields("elasticsearch_indicestats_"+m, jsonParser.Fields, map[string]string{"index_name": "all"}, now)
+	}
+
+	// Individual Indices stats
+	for id, index := range indicesStats.Indices {
+		indexTag := map[string]string{"index_name": id}
+		stats := map[string]interface{}{
+			"primaries": index.Primaries,
+			"total":     index.Total,
+		}
+		for m, s := range stats {
+			f := jsonparser.JSONFlattener{}
+			// parse Json, getting strings and bools
+			err := f.FullFlattenJSON("", s, true, true)
+			if err != nil {
+				return err
+			}
+			acc.AddFields("elasticsearch_indicestats_"+m, f.Fields, indexTag, now)
+		}
+
+		if e.ShardsStats {
+			for shardNumber, shard := range index.Shards {
+
+				// Get Shard Stats
+				flattened := jsonparser.JSONFlattener{}
+				err := flattened.FullFlattenJSON("", shard[0], true, true)
+				if err != nil {
+					return err
+				}
+
+				// determine shard tag and primary/replica designation
+				shardType := "replica"
+				if flattened.Fields["routing_primary"] == true {
+					shardType = "primary"
+				}
+
+				shardTags := map[string]string{
+					"index_name": id,
+					"node_name":  flattened.Fields["routing_node"].(string),
```

Comment:
Panic resolved.
We could try to match it to a host yes, but I don't think we really want it.
Cause you can have 2 nodes on the same host, but not being the same node ... So :)

Reply:
The reason I suggest using the hostname is because we use it as the "node_name" in some of the other measurements and I suspect the times you would have 2 nodes on a host is probably limited to testing environments.  Maybe we should just leave them out though, is there a good way to go from these strings to the hostname?

Reply:
Guess we better rename it to node_id indeed!

Gathering the Name would be possible, but you then need to always gather the clusterwide nodeStats, and there you have a mapping node_id <-> node_name.
Don't know if its worth the (sometimes) additional call?

Reply:
Yeah let's skip this for now, `node_id` will work nicely.